src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -567,16 +567,19 @@ bool System.Collections.IList.Contains(Object item)
             return new Enumerator(this);
         }
 
+
         /// <internalonly/>
         IEnumerator<T> IEnumerable<T>.GetEnumerator() {
-            var enumerator = ObjectPool<PooledIEnumerator>.Shared.Rent();
-            return enumerator.Initalize(this);
+            var enumerator = ObjectPool<PooledIEnumerator<T, Enumerator, List<T>>>.Shared.Rent(
+                                () => new PooledIEnumerator<T, Enumerator, List<T>>());
+            return enumerator.Initalize((list) => new Enumerator(list), this);
```

Comment:
This seems overly complicated. `enumerator._enumerator = new Enumerator(list)` would do the job.